packages/wallets/backpack/src/adapter.ts

```diff
@@ -0,0 +1,259 @@
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection?: Connection,
+        options?: SendTransactionOptions
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            const { signers, ...sendOptions } = options ? options : { signers: undefined };
+
+            const resp = await wallet.send(transaction, signers, sendOptions, connection);
+            if (!resp) {
+                throw new WalletWindowClosedError();
+            }
+            return resp;
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    async signTransaction(transaction: Transaction): Promise<Transaction> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                return (await wallet.signTransaction(transaction)) || transaction;
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    async signAllTransactions(transactions: Transaction[]): Promise<Transaction[]> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                return (await wallet.signAllTransactions(transactions)) || transactions;
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    async signMessage(message: Uint8Array): Promise<Uint8Array> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                const signature = await wallet.signMessage(message);
+                if (!signature) {
+                    throw new Error('User denied signature request');
+                }
```

Comment:
Ah, I see why this could be `null`. But it's probably generally better if the wallet throws an error than just returning nothing, right?

Reply:
It returns null when the user denies the request.  My thinking was to not throw an error for a desired return value.

Reply:
I'm not sure what the idiomatic thing is in javascript, but in rust, this would be an enum return value not an error.

Reply:
Note that we do this for send, and signTx as well.

Reply:
Hmm, it's kind of a judgment call, and ultimately we can adapt to whatever Backpack's behavior is. But I'd say when you have something that returns a Promise, you usually expect it to resolve with the thing you want, or fail. The user declining is a likely expected failure mode, so every downstream caller just has to check for null anyway, and also check for errors.

Reply:
FWIW, I view the `connect` behavior as the same. If the user declines the connection, returning `null` or any value doesn't seem more logical than throwing.

Reply:
Changing to throw per comment here https://github.com/solana-labs/wallet-adapter/pull/481#discussion_r928177693

Reply:
Nulls removed!